src/debug_utils.cc

```diff
@@ -93,42 +93,112 @@ NativeSymbolDebuggingContext::New() {
 
 #else  // __POSIX__
 
+
 class Win32SymbolDebuggingContext final : public NativeSymbolDebuggingContext {
  public:
-  Win32SymbolDebuggingContext() {
+  explicit Win32SymbolDebuggingContext(void* out) {
+    out_ = reinterpret_cast<FILE*>(out);
     current_process_ = GetCurrentProcess();
     USE(SymInitialize(current_process_, nullptr, true));
   }
 
-  ~Win32SymbolDebuggingContext() {
+  ~Win32SymbolDebuggingContext() override {
     USE(SymCleanup(current_process_));
   }
 
-  SymbolInfo LookupSymbol(void* address) override {
-    // Ref: https://msdn.microsoft.com/en-en/library/windows/desktop/ms680578(v=vs.85).aspx
-    char info_buf[sizeof(SYMBOL_INFO) + MAX_SYM_NAME];
-    SYMBOL_INFO* info = reinterpret_cast<SYMBOL_INFO*>(info_buf);
-    char demangled[MAX_SYM_NAME];
+  using NameAndDisplacement = std::pair<std::string, DWORD64>;
+  NameAndDisplacement WrappedSymFromAddr(DWORD64 dwAddress) const {
+    HANDLE hProcess = current_process_;
+
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
+    // Patches:
+    // * Use `fprintf(out_, ` instead of `printf`
+    // * `sym.filename = pSymbol->Name` on success
```

Comment:
I'd rather keep the comment, AFAIK more comment never hurt...
Also they might be useful since the examples on MSDN change from time to time (for example they changed since the last time this code was lifted, and I had to reverse engineer the patches).

Reply:
> Also they might be useful since the examples on MSDN change from time to time

Woudln’t that be a good reason to drop the comment, since the diff can change on both sides, and so the `Patches:` list can grow out of date?

Reply:
I'll sleep on it.